app/src/main/java/com/example/checkpoint/MainActivity.kt

```diff
@@ -54,6 +41,9 @@ class MainActivity : AppCompatActivity() {
 
     private lateinit var locationPermissionHelper: LocationPermissionHelper
 
+    init {
+        onStart()
+    }
```

Comment:
Careful.  onStart() is a lifecycle function, and is automatically invoked when the application starts up.  It should not be invoked any other time.